Sprint16

VankaGam opened the pull request.

VankaGam committed:
Spint 16/
konec vse zarabotalo

IliaAnK commented on app/src/main/java/com/example/playlistmaker/player/data/PlayerInteractorImpl.kt:
```diff
@@ -0,0 +1,64 @@
+package com.example.playlistmaker.player.data
```
Обязательно:
`Interactor` - это сущность `domain` слоя. Нужно перенести  в соответствующую директорию

IliaAnK commented on app/src/main/java/com/example/playlistmaker/player/data/PlayerInteractorImpl.kt:
```diff
@@ -0,0 +1,64 @@
+package com.example.playlistmaker.player.data
+
+import android.media.MediaPlayer
```
Обязательно:
В `domain` слое не должно быть зависимостей на `Android SDK`.
Нужно вынести работу с плеером в `Repository`. Имплементация репозитория должна находится в `data` млое

IliaAnK commented on app/src/main/java/com/example/playlistmaker/player/data/PlayerInteractorImpl.kt:
```diff
@@ -0,0 +1,64 @@
+import kotlinx.coroutines.delay
+
+class PlayerInteractorImpl : PlayerInteractor {
+    private var player: MediaPlayer? = null
```
Рекомендация:
Это зависимость, которую стоит создавать в `Creator` и передавать через конструктор класса.

IliaAnK commented on app/src/main/java/com/example/playlistmaker/search/data/repository/SearchHistoryRepositoryImpl.kt:
```diff
@@ -1,8 +1,9 @@
+import com.example.playlistmaker.search.domain.repository.SearchHistoryRepository
 
 class SearchHistoryRepositoryImpl(context: Context) : SearchHistoryRepository {
     private val storage = SearchHistoryStorage(context)
```
Рекомендация:
Это зависимость, которую стоит создавать в `Creator` и передавать через конструктор класса.
А контекст для `SearchHistoryStorage` стоит заменить сразу на `SharedPreferences`

IliaAnK commented on app/src/main/java/com/example/playlistmaker/creator/Creator.kt:
```diff
@@ -0,0 +1,45 @@
+        val trackRepo = TrackRepositoryImpl()
+        val historyRepo = SearchHistoryRepositoryImpl(context)
+
+        return SearchViewModel(
```
Обязательно:
`SearchViewModel` необходимо создавать с помощью `ViewModelProvider` также, как и остальные `ViewModel`

VankaGam committed:
Spint 16/
Peredelel obyzatelnie rekomend;
ispravil bag;
nastroil cveta switch(deystvitelno bili ne pravilnie)

IliaAnK commented on app/src/main/java/com/example/playlistmaker/search/data/repository/SearchHistoryRepositoryImpl.kt:
```diff
@@ -1,26 +1,17 @@
+class SearchHistoryRepositoryImpl(
+    private val sharedPreferences: SharedPreferences
+) : SearchHistoryRepository {
+    private val storage = SearchHistoryStorage(sharedPreferences)
```
Рекомендация:
Репозиторий не должен отвечать за создание своих зависимостей.
Нужно перенести создание `SearchHistoryStorage` в `Creator` и передавать его сюда через конструктор класса.

VankaGam committed:
Spint 16/
Popravil. Teper PlayerViewModel rabotaet s PlayerInteractor iz domain
